toolbox/org.lamport.tla.toolbox.tool.prover/src/org/lamport/tla/toolbox/tool/prover/job/ProverJob.java

```diff
@@ -763,6 +804,34 @@ private String[] constructCommand()
         return (String[]) command.toArray(new String[command.size()]);
     }
 
+    /**
+     * Constructs and returns the command to get the prover config.
+     * 
+     * @return
+     */
+    private String[] constructCommandForConfig()
+    {
+
+        List<String> command = new ArrayList<String>();
+        /*
+         * Launch from the command line:
+         * 
+         * > <tlapm-command> --config
+         */
+
+        command.add(tlapmPath.toOSString());
+
+        // WSL hack. See constructCommand above.
+        final boolean useWSL = tlapmPath.toOSString().contains("wsl.exe");
+        if (useWSL) {
+            command.add("tlapm");
+        }
+
+        command.add("--config");
+
+        return (String[]) command.toArray(new String[command.size()]);
```

Comment:
I don't think there's any need for an awkward array conversion here. It should be possible to simplify this to `return command;` if you make the method return `List<String>` instead of `String[]`. The only use of this method is as an argument to `new ProcessBuilder`, and [that constructor can take either `String[]` or `List<String>`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ProcessBuilder.html#%3Cinit%3E(java.util.List)).

Reply:
Right. I just reproduced the constructCommand pattern already present in the same file. I will push a simplified version.